main.js

```diff
@@ -280,364 +716,113 @@ function render2D(foldObj) {
-                isCamRotating = true;
-            } else if (e.button === 2) {
-                isCamPanning = true;
-            }
-        });
-
-        // listen for mouse release on the whole page to prevent accidental sticky rotate
-        document.addEventListener("mouseup", e => {
-            if (isCamRotating || isCamPanning) {
-                isCamRotating = false;
-                isCamPanning = false;
-            }
-        });
-
-        // orbit the camera upon mouse movement in the canvas, pan if right clicked
-        canvas.addEventListener("mousemove", e => {
-            if (isCamRotating) {
-                theta -= degToRad(e.movementX);
-                let n = phi - degToRad(e.movementY);
-                if (n > Math.PI - restrictionRangeY) {
-                    phi = Math.PI - restrictionRangeY;
-                } else if (n < restrictionRangeY) {
-                    phi = restrictionRangeY;
-                } else {
-                    phi = n;
-                }
-            } else if (isCamPanning) { // prevent both pan and rotate at same time
-                origin.setComponent(0, origin.x - (e.movementX * sensitivityScale)); // x axis movement
-                origin.setComponent(2, origin.z - (e.movementY * sensitivityScale));
-            }
-        });
-
-        // use scroll wheel to zoom
-        canvas.addEventListener("wheel", e => {
-            e.preventDefault();
-            let newZoom = rotationRadius + e.deltaY * zoomSensitivity;
-            if (newZoom < zoomLimit) {
-                rotationRadius = zoomLimit;
-            } else if (newZoom > zoomMax) {
-                rotationRadius = zoomMax;
+    // to be called whenever a change is made to selected, updates the buttons to available options
+    function changeSelection() {
+        let np = selectedPoints.length;
+        let nl = selectedLines.length;
+        for (let i = 0; i < buttons.length; i++) {
+            if (huzitaOptions[i].p === np && huzitaOptions[i].l === nl) {
+                buttons[i].disabled = false;
             } else {
-                rotationRadius = newZoom;
+                buttons[i].disabled = true;
             }
-        });
-
-        if (!reRender) {
-            // prevent right clicks from opening the context menu anywhere
-            document.addEventListener("contextmenu", e => {
-                e.preventDefault();
-            });
         }
     }
 
-    function animate() {
-        requestAnimationFrame(animate);
-
-        let sphCoords = new THREE.Spherical(rotationRadius, phi, theta);
-        camera.position.setFromSpherical(sphCoords);
-
-        // offset spherical position to new origin as well for panning
-        camera.position.addVectors(camera.position, origin);
-        camera.lookAt(origin);
-
-        renderer.render(scene, camera);
+    function foldAxiom1() {
+        createNewEdge(mathVertices[selectedPoints[0]], mathVertices[selectedPoints[1]]);
```

Comment:
I _believe_ this can potentially create new points at intersections too.

Reply:
Check my comment replying to the one about `foldAxiom2`